### DISCUSSION TITLE
Course creation question

### DISCUSSION THREAD

**Discussion by sinhalaquiz** [kantord/LibreLingo#1737] [General]
Hi,

Thank you for starting this project.

This is a quick introduction while I start exploring the project in a bit more detail.

I have been searching for a platform to host a [course that I've been building](https://app.memrise.com/course/5640090/sinhalese-with-native-script/) for some time. I was hoping to get the content through to Duolingo but alas now they've decided to drop the volunteer program so there's no chance of that now :-(

Anyway, I was searching for a Memrise alternative and your project came up. So I thought I might give it a spin and see. I realise that you currently don't support non Latin character sets - but maybe I might be able to give you any assistance you need. I have some coding experience.

Just a quick question. I noticed that there are scripts pulling audio from an external source. Do you support audio included as part of the course as well?

**Comment by kantord** [comment]
> I realise that you currently don't support non Latin character sets - but maybe I might be able to give you any assistance you need. I have some coding experience.

exactly. I think if you start creating the course now, nothing will crash probably. The issue could rather be the input methods, and the lack of specific features to teach the writing system itself

**Reply by sinhalaquiz** [reply]
> ... and the lack of specific features to teach the writing system itself

Could it not be like words in a Skill?

For example see the Alphabet Skills in my SInhala course https://app.memrise.com/course/5640090/sinhalese-with-native-script/1/
If you register and try it out, you can see it shows the characters, pronunciation in English as well as an audio clip with the sound.

**Comment by kantord** [comment]
> Do you support audio included as part of the course as well?

Not really. Technically we could support this, however I would not recommend relying on a feature like this. Instead of implementing such features in LibreLingo itself, I'd rather like to rely on other platforms for uploading audios. The reason is because LibreLingo is not meant as a platform for uploading/hosting audios. Rather we should rely on other platforms for this, at least as long as your course content is open source/public domain/etc. 

There are a couple issues already open for this:

- https://github.com/kantord/LibreLingo/issues/1464
- https://github.com/kantord/LibreLingo/issues/714

As a first step however, **you can start creating a course without audios** and upload/generate audios for it later!

**Comment by sinhalaquiz** [comment]
Hi again.

Having some difficulty setting up the dev environment (I'm not familiar with nodejs so please bear with me)

I'm following this document: https://librelingo.app/docs/#setting-up-the-development-environment

The bit I'm having trouble is with the web app.

I installed yarn as described here: https://linuxize.com/post/how-to-install-yarn-on-ubuntu-20-04/

Here's the output when running yarn:
```

$ yarn set version classic
Resolving classic to a url...
Downloading https://github.com/yarnpkg/yarn/releases/download/v1.22.17/yarn-1.22.17.js...
Saving it into /home/XXX/src/LibreLingo/.yarn/releases/yarn-1.22.17.cjs...
Updating /home/XXX/src/LibreLingo/.yarnrc...
Done!

$ yarn
yarn install v1.22.17
[1/4] Resolving packages...
[2/4] Fetching packages...
error @typescript-eslint/eslint-plugin@5.2.0: The engine "node" is incompatible with this module. Expected version "^12.22.0 || ^14.17.0 || >=16.0.0". Got "10.19.0"
error Found incompatible module.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.

```
Any suggestions?

Thank you

**Reply by kantord** [reply]
Looks like your version of node is very old. You should try updating to a newer version or you can try using `nvm` in order to use a different version of node just for LibreLingo. 

More info about nvm: https://github.com/nvm-sh/nvm

You can also consult the documentation of your operating system to find how to update your node version

**Reply by sinhalaquiz** [reply]
Thanks. Got the app running. See the following output:

```

500

Cannot find module './spanish-from-english/courseData.json'

webpackAsyncContext/<@http://localhost:3000/client/425c6e6fd939d0a9446f/course_$courseName.course_$courseName.js:2620:12
```
Do I need to export a course first for it to work?

**Reply by kantord** [reply]
> Do I need to export a course first for it to work?

yes, currently there's no way to avoid that.

**Reply by sinhalaquiz** [reply]
ok... where exactly is the spanish-from-english course? I'm working off trunk and I only see the following:

```
courses/basque-from-english 
courses/test
temporarily_inactive_courses/bangla-from-english
temporarily_inactive_courses/french-from-english
temporarily_inactive_courses/basque-from-english
temporarily_inactive_courses/parsig-from-english
```
I'm assuming I need the spanish-from-english course because that's the link in the front page..

**Reply by kantord** [reply]
Sorry for the late reply. 

Please try running

```
yarn web installAllExternalCourses
```

also, do you have any suggestion how this could be a lot more user friendly?

**Reply by sinhalaquiz** [reply]
At this stage, adding that line to the dev docs would do the trick. For me the hardest part was trying to guess my way around issues.

Perhaps a bootstrap.sh in the Scripts section?
